content/en/docs/instrumentation/swift/manual.md

```diff
@@ -0,0 +1,248 @@
+                             .startSpan() //automatically captures `active span` as parent
+  // do work
+  childSpan.end()
+}
+
+```
+
+### Getting the Current Span
+
+Sometimes it's useful to do something with the current/active span. Here's how to access the current span from an arbitrary point in your code.
+
+```swift 
+  let currentSpan = OpenTelemetry.instance.contextProvider.activeSpan
+```
+### Span Attributes
+
+Spans can also be annotated with additional attributes. All spans will be automatically annotated with the `Resource` attributes attached to the tracer provider.
+The Opentelementry-swift sdk already provides instrumentation of common attributes in the `SDKResourceExtension` instrumentation.
+In this example a span for a network request capturing details about that request using existing [semantic conventions](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/).
+
+```swift
+let span = tracer.spanBuilder("/resource/path").startSpan()
+span.setAttribute("http.method", "GET");
+span.setAttribute("http.url", url.toString());
+```
+
+### Creating Span Events
+A Span Event can be thought of as a structured log message (or annotation) on a Span, typically used to denote a meaningful, singular point in time during the Span’s duration.
+
+```swift
+            let attributes = [
+                "key" : AttributeValue.string("value"),
+                "result" : AttributeValue.int(100)
+            ]
+            span.addEvent(name: "computation complete", attributes: attributes)
+```
+
+### Setting Span Status
+A [status](https://opentelemetry.io/docs/concepts/signals/traces/#span-status) can be set on a span, typically used to specify that a span has not completed successfully - SpanStatus.Error. In rare scenarios, you could override the Error status with OK, but don’t set OK on successfully-completed spans.
+
+The status can be set at any time before the span is finished:
+
+```swift
+func myFunction() {
+  let span = someTracer.spanBuilder(spanName: "my span").startSpan() 
+  defer {
+    span.end()
+  }
+  guard let criticalData = get() else {
+      span.status = .error(description: "something bad happened")
+      return 
+  }
+  // do somthing 
+}
+```
+### Recording exceptions in Spans
+Semantic conventions provide special demarcation for events that record exceptions:
+```swift
+  let span = someTracer.spanBuilder(spanName: "my span").startSpan() 
+do {
+  try throwingFunction() 
+} catch {
+  span.addEvent(name: SemanticAttributes.exception.rawValue,
+    attributes: [SemanticAttributes.exceptionType.rawValue: AttributeValue.string(String(describing: type(of: error))),
+                 SemanticAttributes.exceptionEscaped.rawValue: AttributeValue.bool(false),
+                 SemanticAttributes.exceptionMessage.rawValue: AttributeValue.string(error.localizedDescription)])
+  })
+  span.status = .error(description: error.localizedDescription)
+}
+span.end()
+```
+## Metrics 
```

Comment:
We should skip this section until metrics are good to go.

Reply:
Yes, agreed